application.go

```diff
@@ -0,0 +1,34 @@
+package dogma
+
+// Application is an interface implemented by the application and used by the
+// engine to describe the structure of an application.
+type Application interface {
+	// Configure configures the behavior of the engine as it relates to this
+	// application.
+	//
+	// c provides access to the various configuration options, such as specifying
+	// which message handlers the application contains.
+	Configure(c ApplicationConfigurer)
+}
+
+// ApplicationConfigurer is an interface implemented by the engine and used by
+// the application to configure options related to the application itself.
+//
+// It is passed to Application.Configure(), typically upon initialization of the
+// engine.
+type ApplicationConfigurer interface {
+	// Name sets the name of the application.
+	Name(n string)
+
+	// RegisterAggregate configures the engine to route messages to h.
+	RegisterAggregate(h AggregateMessageHandler)
+
+	// RegisterProcess configures the engine to route messages to h.
```

Comment:
The `h` isn't specified in the method signature?
Same for the next couple.

Reply:
Fixed.